### PR TITLE
switch to mariadb 10.1 client + assorted cleanups

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by jhoblitt** [comment]
@athornton This isn't a complete cleanup but it does clean things up a bit and make some improvements.  Could you give this a quick look?

File: run.sh [hunk 1]
```diff
@@ -3,51 +3,57 @@
...
+  echo -e $@
+}
+
+# use UTC for timestamps
+datecmd() {
+  # shellcheck disable=SC2068
+  date --utc $@
+}
+
+[[ -z "${AWS_ACCESS_KEY_ID+x}" ]] &&
```
Comment (athornton) [inline run.sh]: I just tested it, and it clearly works, but how does the "+x" construct work?  I would just do -z "${VAR}", which, with the double-quotes, works too.

File: run.sh [hunk 2]
```diff
@@ -3,51 +3,57 @@
...
+  fail "You need to set the AWS_ACCESS_KEY_ID environment variable."
+
+[[ -z "${AWS_SECRET_ACCESS_KEY+x}" ]] &&
```
Comment (athornton) [inline run.sh]: You could be all fancy and loop over the var names you want to set, but this is probably clearer.

**Review by athornton — APPROVED** [review]

File: run.sh (hunk 1 above)
Comment (jhoblitt) [reply run.sh]: It isn't the same test.  The `${foo+<anything>}` in a posix-ism to test if the variable has been defined.  Without this, you couldn't set the value to an empty string (`""`).

https://unix.stackexchange.com/questions/56837/how-to-test-if-a-variable-is-defined-at-all-in-bash-prior-to-version-4-2-with-th

File: run.sh (hunk 2 above)
Comment (jhoblitt) [reply run.sh]: I thought about putting the var name + error message into a hash but was concerned that might lessen that chance of this being accepted back upstream.

File: run.sh [hunk 3]
```diff
@@ -20,29 +20,24 @@ datecmd() {
...
+  [AWS_SECRET_ACCESS_KEY]='You need to set the AWS_SECRET_ACCESS_KEY environment variable.'
+  [AWS_BUCKET]='You need to set the AWS_BUCKET environment variable.'
+  [PREFIX]='You need to set the PREFIX environment variable.'
+  [MYSQL_ENV_MYSQL_USER]='You need to set the MYSQL_ENV_MYSQL_USER environment variable.'
+  [MYSQL_ENV_MYSQL_PASSWORD]='You need to set the MYSQL_ENV_MYSQL_PASSWORD environment variable.'
+  [MYSQL_PORT_3306_TCP_ADDR]='You need to set the MYSQL_PORT_3306_TCP_ADDR environment variable or link to a container named MYSQL.'
+  [MYSQL_PORT_3306_TCP_PORT]='You need to set the MYSQL_PORT_3306_TCP_PORT environment variable or link to a container named MYSQL.'
+)
+
+for k in "${!req_vars[@]}"; do
```
Comment (athornton) [inline run.sh]: I like it.  I'd hate to be the poor person who is sent to maintain it with no prior knowledge of what's going on, though.

**Review by athornton — APPROVED** [review]

File: run.sh (hunk 3 above)
Comment (jhoblitt) [reply run.sh]: I agree that associative arrays aren't a common bash feature but I think what is going on here is fairly obvious.  It is certainly DRYer than cut'n'pasting a test for each variable.

**Comment by jhoblitt** [comment]
@athornton I've fixed the last of the original shellcheck warnings and add a couple of small features.  Could you give it another quick look?

**Comment by jhoblitt** [comment]
Tested as working:
```
# docker run -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
>   -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
>   -e AWS_BUCKET=jhoblitt-test-dump \
>   -e PREFIX=mysqldump \
>   -e MYSQL_ENV_MYSQL_USER=admin \
>   -e MYSQL_ENV_MYSQL_PASSWORD=$MYSQL_ENV_MYSQL_PASSWORD \
>   -e MYSQL_PORT_3306_TCP_ADDR=jenkins-prod-qadb.lsst.codes \
>   -e MYSQL_PORT_3306_TCP_PORT=3306 \
>   -e CREATE_AWS_BUCKET=true \
>   -e DUMP_NAME=latest2 \
>   lsstsqre/mysqldump_to_s3
Creating S3 bucket s3://jhoblitt-test-dump...
make_bucket: jhoblitt-test-dump
Starting dump of --all-databases database(s) from jenkins-prod-qadb.lsst.codes...
Dumped to s3://jhoblitt-test-dump/mysqldump/latest2.sql.gz...
2017-02-07 21:27:49  954.5 KiB latest2.sql.gz
Done!
```

**Pull request merged**